src/main/java/com/rrsys/productsapi/services/impl/ProductServiceImpl.java

```diff
@@ -46,28 +46,21 @@ public ProductsEntity update(UUID id,ProductsEntity newValues) {
         //Alterando novo dado
         productDb.setDescription(newValues.getDescription());
         productDb.setName(newValues.getName());
-        productDb.setAmount(newValues.getAmount());
 
-        productPercent(id,newValues);
+        //Verificação se o novo valor que for passado foi acima de 51%, é lançada uma exceçao
+        double percent = productDb.getAmount() / 100 * 51;
+        if (newValues.getAmount() > (percent + productDb.getAmount())) {
+            throw new RuntimeException("Product not accept");
+        } else {
```

Comment:
remove o else pois nesse caso de o if for true é lançado um exception
então não vai executar a próxima linha, se caso for false a linha seguinte é executada mudando o valor do produto
```suggestion
        } 
```